src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +164,213 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
                         idsToDelete.Add(queueItem.DocumentId);
                 }
             }
+        }
+        catch (ProduceException<string, byte[]> e)
+        {
+            if (e.Error.Code == ErrorCode.TopicAuthorizationFailed)
+            {
+                var topicErrorMessageBuilder = new StringBuilder(
+                    $"ETL process: {Name}. Failed to produce message.{Environment.NewLine}" +
+                    $"Add an required ACL permissions to your API Key that will make WRITE operations on the '{currentTopic.Name}' topic possible.");
+                
+                TryAbortTransactionAfterException(topicErrorMessageBuilder, out _);
+                throw new QueueLoadException(topicErrorMessageBuilder.ToString(), e);
+            }
 
-            _producer.CommitTransaction();
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Produce message operation failed. ");
+            if (TryToGetAclErrorsDetails(out var detailsMessage, currentTopic.Name))
+                errorMessageBuilder.Append(detailsMessage);
+            
+            TryAbortTransactionAfterException(errorMessageBuilder, out _);
+            throw new QueueLoadException(errorMessageBuilder.ToString(), e);
         }
         catch (Exception ex)
         {
-            try
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to produce message.{Environment.NewLine}");
+            TryAbortTransactionAfterException(errorMessageBuilder, out _);
+            throw new QueueLoadException(ex.Message, ex);
+        }
+        
+        
+        try
+        {
+            _producer.CommitTransaction();
+        }
+        catch (KafkaTxnRequiresAbortException ex)
+        {
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Commit transaction failed, due to {nameof(KafkaTxnRequiresAbortException)}. ");
```

Comment:
`due to {nameof(KafkaTxnRequiresAbortException)}` -> this is not needed. We'll have the actual exception anyway.

Reply:
done